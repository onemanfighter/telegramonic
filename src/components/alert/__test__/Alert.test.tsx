import { render } from '@testing-library/react';
import AlertComponent from '../Alert';

describe('Alert', () => {
  it('should render correctly', () => {
    const { container } = render(
      <AlertComponent title="title" description="description" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with status', () => {
    const { container } = render(
      <AlertComponent title="title" description="description" status="error" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with variant', () => {
    const { container } = render(
      <AlertComponent
        title="title"
        description="description"
        variant="solid"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly without icon', () => {
    const { container } = render(
      <AlertComponent
        title="title"
        description="description"
        showIcon={false}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with status and variant', () => {
    const { container } = render(
      <AlertComponent
        title="title"
        description="description"
        status="error"
        variant="solid"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with all props', () => {
    const { container } = render(
      <AlertComponent
        title="title"
        description="description"
        status="error"
        variant="solid"
        showIcon={false}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with loading status', () => {
    const { container } = render(
      <AlertComponent
        title="title"
        description="description"
        status="loading"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
