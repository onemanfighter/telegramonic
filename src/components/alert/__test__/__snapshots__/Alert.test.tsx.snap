// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly 1`] = `
<div>
  <div
    class="chakra-alert css-1hwd9xd"
    data-status="info"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-1c6vscd"
      data-status="info"
    >
      <svg
        class="chakra-icon css-qk6lof"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="chakra-alert__title css-0"
      data-status="info"
    >
      title
    </div>
    <div
      class="chakra-alert__desc css-6n7j50"
      data-status="info"
    >
      description
    </div>
  </div>
</div>
`;

exports[`Alert should render correctly with all props 1`] = `
<div>
  <div
    class="chakra-alert css-1hwd9xd"
    data-status="error"
    role="alert"
  >
    <div
      class="chakra-alert__title css-0"
      data-status="error"
    >
      title
    </div>
    <div
      class="chakra-alert__desc css-6n7j50"
      data-status="error"
    >
      description
    </div>
  </div>
</div>
`;

exports[`Alert should render correctly with loading status 1`] = `
<div>
  <div
    class="chakra-alert css-1hwd9xd"
    data-status="loading"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-1c6vscd"
      data-status="loading"
    >
      <div
        class="chakra-spinner css-1e4oo9u"
      >
        <span
          class="css-8b45rq"
        >
          Loading...
        </span>
      </div>
    </span>
    <div
      class="chakra-alert__title css-0"
      data-status="loading"
    >
      title
    </div>
    <div
      class="chakra-alert__desc css-6n7j50"
      data-status="loading"
    >
      description
    </div>
  </div>
</div>
`;

exports[`Alert should render correctly with status 1`] = `
<div>
  <div
    class="chakra-alert css-1hwd9xd"
    data-status="error"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-1c6vscd"
      data-status="error"
    >
      <svg
        class="chakra-icon css-qk6lof"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="chakra-alert__title css-0"
      data-status="error"
    >
      title
    </div>
    <div
      class="chakra-alert__desc css-6n7j50"
      data-status="error"
    >
      description
    </div>
  </div>
</div>
`;

exports[`Alert should render correctly with status and variant 1`] = `
<div>
  <div
    class="chakra-alert css-1hwd9xd"
    data-status="error"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-1c6vscd"
      data-status="error"
    >
      <svg
        class="chakra-icon css-qk6lof"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="chakra-alert__title css-0"
      data-status="error"
    >
      title
    </div>
    <div
      class="chakra-alert__desc css-6n7j50"
      data-status="error"
    >
      description
    </div>
  </div>
</div>
`;

exports[`Alert should render correctly with variant 1`] = `
<div>
  <div
    class="chakra-alert css-1hwd9xd"
    data-status="info"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-1c6vscd"
      data-status="info"
    >
      <svg
        class="chakra-icon css-qk6lof"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="chakra-alert__title css-0"
      data-status="info"
    >
      title
    </div>
    <div
      class="chakra-alert__desc css-6n7j50"
      data-status="info"
    >
      description
    </div>
  </div>
</div>
`;

exports[`Alert should render correctly without icon 1`] = `
<div>
  <div
    class="chakra-alert css-1hwd9xd"
    data-status="info"
    role="alert"
  >
    <div
      class="chakra-alert__title css-0"
      data-status="info"
    >
      title
    </div>
    <div
      class="chakra-alert__desc css-6n7j50"
      data-status="info"
    >
      description
    </div>
  </div>
</div>
`;
